### PR TITLE
add handling for corrupt files during sync

### PR DESCRIPTION
**Pull request by earthman1**

### CODE REVIEW

File: cloudsync/sync/manager.py [hunk 1]
```diff
@@ -1536,10 +1568,17 @@ def handle_hash_diff(self, sync, changed, synced):
 
         assert sync[synced].oid
 
-        if not self.download_changed(changed, sync):
+        try:
+            dc_result = self.download_changed(changed, sync)
```
Comment (earonesty) [inline cloudsync/sync/manager.py:1572]: variable not needed, up to u

File: cloudsync/sync/state.py [hunk 2]
```diff
@@ -107,16 +110,28 @@ def __setattr__(self, k, v):
             object.__setattr__(self, k, v)
             return
 
+        if k == "exists":
+            if v == CORRUPT and not self.is_corrupt:
+                self._saved_exists = self.exists
+                self._exists = v
+                return
+            if v != CORRUPT and self.is_corrupt:
+                self._saved_exists = self._translate_exists(v)
```
Comment (earonesty) [inline cloudsync/sync/state.py:119]: if the caller is setting it to exists = False or exists = Missing, it should probably be allowed, right?

File: cloudsync/sync/state.py [hunk 3]
```diff
@@ -107,16 +110,28 @@ def __setattr__(self, k, v):
             object.__setattr__(self, k, v)
             return
 
+        if k == "exists":
+            if v == CORRUPT and not self.is_corrupt:
+                self._saved_exists = self.exists
+                self._exists = v
```
Comment (earonesty) [inline cloudsync/sync/state.py:116]: we don't want to call parent update.... why?

File: cloudsync/sync/state.py [hunk 4]
```diff
@@ -182,6 +200,58 @@ def clean_temp(self):
...
+
+    @property
+    def corrupt_exists(self):
+        return self.is_corrupt and self._saved_exists == EXISTS
+
+    @property
+    def corrupt_gone(self):
+        return self.is_corrupt and self._saved_exists in (TRASHED, MISSING, LIKELY_TRASHED)
+
+    def serialize(self) -> dict:
```
Comment (earonesty) [inline cloudsync/sync/state.py:215]: nice

File: cloudsync/sync/state.py (hunk 2 above)
Comment (earthman1) [reply cloudsync/sync/state.py:119]: I can create an "uncorrupt" method... comments are also good

File: cloudsync/sync/state.py (hunk 3 above)
Comment (earthman1) [reply cloudsync/sync/state.py:116]: because nothing actually got updated

File: cloudsync/sync/state.py (hunk 4 above)
Comment (earthman1) [reply cloudsync/sync/state.py:215]: thx

File: cloudsync/sync/state.py (hunk 2 above)
Comment (earthman1) [reply cloudsync/sync/state.py:120]: done

**Review by earonesty — APPROVED** [review]

**Pull request merged**